### PR TITLE
events: turn event definitions instantiable 

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
commit ce65c39a2 (HEAD -> myowntracee)
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Thu Jun 15 01:58:08 2023

    events: turn definitions instantiable for extensions
    
    WIP
    
    related: #3170

commit 366347403
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Wed Jun 14 21:42:40 2023

    chore: spelling and comments

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
I am missing unit tests fixes, full functional tests and to implement unit tests for all types added to the "events" package (and probably rename events to "definitions" or something that does not make it look like a tracee event.

File: pkg/policy/policy_file_test.go
Comment (rafaeldtinoco) [inline pkg/policy/policy_file_test.go:1]: TODO: needs fixing

File: pkg/events/events_test.go
Comment (rafaeldtinoco) [inline pkg/events/events_test.go:1]: TODO: needs fixing

**Comment by NDStrahilevitz** [comment]
Would like to review this one once stable.

**Comment by rafaeldtinoco** [comment]
Im fixing some stuff observed in unit testing: Golang does not like eventual consistency by RCU approach using atomics for slices, for example. The pointer dereferences are atomic but runtime causes races when copying the slices. 

Also, Passing slices of values is also problematic (like AddEvents) because the slice reads aren't protected by an external mutexes, for example. So there could only exist AddEvent() and iteration done outside the logic. 

By benchmarking atomic approach and read/write mutexes, I could also see there is little benefit in adding extra complex logic, so I'll keep it simpler but all methods thread-safe.

File: pkg/events/usermode.go [hunk 1]
```diff
@@ -92,30 +100,30 @@ func fetchInitNamespaces() map[string]uint32 {
...
-			{ArgMeta: def.Params[1], Value: info.Container.ContainerId},
-			{ArgMeta: def.Params[2], Value: info.Ctime.UnixNano()},
-			{ArgMeta: def.Params[3], Value: container.Image},
-			{ArgMeta: def.Params[3], Value: container.ImageDigest},
-			{ArgMeta: def.Params[4], Value: container.Name},
-			{ArgMeta: def.Params[5], Value: container.Pod.Name},
-			{ArgMeta: def.Params[6], Value: container.Pod.Namespace},
-			{ArgMeta: def.Params[7], Value: container.Pod.UID},
-			{ArgMeta: def.Params[8], Value: container.Pod.Sandbox},
+			{ArgMeta: def.GetParams()[0], Value: info.Runtime.String()},
```
Comment (rafaeldtinoco) [inline pkg/events/usermode.go:111]: FIX: call getparams a single time (perf reasons).

**Pull request closed without merging**